cranelift-codegen/meta/src/cdsl/xform.rs

```diff
@@ -138,6 +138,28 @@ impl Transform {
     }
 }
 
+/// Inserts, if not present, a name in the `symbol_table`. Then returns its index in the variable
+/// pool `var_pool`. If the variable was not present in the symbol table, then add it to the list of
+/// `defined_vars`.
+fn var_index(
+    name: &'static str,
+    symbol_table: &mut SymbolTable,
+    defined_vars: &mut Vec<VarIndex>,
```

Comment:
I think the current scheme is the simplest. Returning a boolean to then branch on it would be harder to read from my point of view.

Note, I could replace the code from the next commit to be less alien by replacing:
```rust
            let mut temp_vars = vec![];
            let var_index = var_index(var.name, symbol_table, &mut temp_vars, var_pool);
            var_pool.get_mut(var_index).set_def(position, def_index);
            block_pool.create_block(var_index, def_index);
            assert!(
                temp_vars.is_empty(),
                format!(
                    "Block {} was not registered during the first visit",
                    var.name
                )
            );
```

by

```rust
            let nb_defined_vars = defined_vars.len();
            let var_index = var_index(var.name, symbol_table, defined_vars, var_pool);
            var_pool.get_mut(var_index).set_def(position, def_index);
            block_pool.create_block(var_index, def_index);
            assert!(
                defined_vars.len() == nb_defined_vars,
                format!(
                    "Block {} was not registered during the first visit",
                    var.name
                )
            );
```

We could also completely replace the `var_index` call by:

```
            let var_index = symbol_table.get(var.name).expect(
                format!(
                    "Block {} was not registered during the first visit",
                    var.name
                )
            );
            var_pool.get_mut(var_index).set_def(position, def_index);
            block_pool.create_block(var_index, def_index);
```

Since the first loop should have registered all new block names.

Reply:
> We could also completely replace the var_index call by:

Yes, that looks more straight to the point, thanks.